lib/address-derivation-discovery/lib/Cardano/Wallet/Address/Derivation/SharedKey.hs

```diff
@@ -147,6 +147,7 @@ constructAddressFromIx role pTemplate dTemplate ix =
         Nothing ->
             createEnterpriseAddress pScript
 
+-- | NOTE: The roles 'DRep', 'CCCold', 'CCHot' are not supported.
```

Comment:
Although, I think restricting the domain with a new type containing a subset of `CA.Role` should be fairly painless. May try...